yarn-project/circuit-types/src/l2_block.ts

```diff
@@ -603,27 +317,17 @@ export class L2Block {
   getTx(txIndex: number) {
     this.assertIndexInRange(txIndex);
 
-    const newCommitments = this.newCommitments
-      .slice(MAX_NEW_COMMITMENTS_PER_TX * txIndex, MAX_NEW_COMMITMENTS_PER_TX * (txIndex + 1))
-      .filter(x => !x.isZero());
-    const newNullifiers = this.newNullifiers
-      .slice(MAX_NEW_NULLIFIERS_PER_TX * txIndex, MAX_NEW_NULLIFIERS_PER_TX * (txIndex + 1))
-      .filter(x => !x.isZero());
-    const newPublicDataWrites = this.newPublicDataWrites
-      .slice(MAX_PUBLIC_DATA_UPDATE_REQUESTS_PER_TX * txIndex, MAX_PUBLIC_DATA_UPDATE_REQUESTS_PER_TX * (txIndex + 1))
-      .filter(x => !x.isEmpty());
-    const newL2ToL1Msgs = this.newL2ToL1Msgs
-      .slice(MAX_NEW_L2_TO_L1_MSGS_PER_TX * txIndex, MAX_NEW_L2_TO_L1_MSGS_PER_TX * (txIndex + 1))
-      .filter(x => !x.isZero());
-    const newContracts = this.newContracts
-      .slice(MAX_NEW_CONTRACTS_PER_TX * txIndex, MAX_NEW_CONTRACTS_PER_TX * (txIndex + 1))
-      .filter(x => !x.isZero());
-    const newContractData = this.newContractData
-      .slice(MAX_NEW_CONTRACTS_PER_TX * txIndex, MAX_NEW_CONTRACTS_PER_TX * (txIndex + 1))
-      .filter(x => !x.isEmpty());
+    const txEffect = this.body.txEffects[txIndex];
+
+    const newNoteHashes = txEffect.newNoteHashes.filter(x => !x.isZero());
+    const newNullifiers = txEffect.newNullifiers.filter(x => !x.isZero());
+    const newPublicDataWrites = txEffect.newPublicDataWrites.filter(x => !x.isEmpty());
+    const newL2ToL1Msgs = txEffect.newL2ToL1Msgs.filter(x => !x.isZero());
+    const newContracts = txEffect.contractLeaves.filter(x => !x.isZero());
+    const newContractData = txEffect.contractData.filter(x => !x.isEmpty());
```

Comment:
Or isEmpty/isZero naming became inconsistent. Created an [issue](https://github.com/AztecProtocol/aztec-packages/issues/4638) for this.